include/circt/Dialect/OM/OMOpInterfaces.td

```diff
@@ -15,6 +15,102 @@
 
 include "mlir/IR/OpBase.td"
 
+def ClassFieldsLike : OpInterface<"ClassFieldsLike"> {
+  let cppNamespace = "circt::om";
+
+  let description = [{
+    Marks the end of a class-like definition by providing a list of accessible
+    fields.
+  }];
+  let methods = [
+    InterfaceMethod<"Get the class-like field to type map",
+      "std::optional<mlir::Type>", "getFieldType", (ins "mlir::StringAttr":$name)>,
+    InterfaceMethod<"Print class fields like",
```

Comment:
I think  `printClassFieldsLike` is not suitable as OpInterfaceMethod since this is not useful other than MLIR prtiing. Could you move this as a static function in OMOps.cpp?